File: myutils/src/test/java/pl/mareklangiewicz/myutils/MyMathUtilsTest.java
package pl.mareklangiewicz.myutils;

import com.noveogroup.android.log.Logger;
import com.noveogroup.android.log.MyHandler;
import com.noveogroup.android.log.MyLogger;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

/**
 * Created by Marek Langiewicz on 04.11.15.
 */
public class MyMathUtilsTest {

    private static final MyLogger log = new MyLogger("UT");

    @Before
    public void setUp() throws Exception {
        MyHandler.sPrintLnLevel = Logger.Level.VERBOSE;
    }

    @After
    public void tearDown() throws Exception {

    }

    @Test
    public void testScale0d() throws Exception {

    }

    @Test
    public void testScale1d() throws Exception {

    }

    @Test
    public void testScale2d() throws Exception {

    }

    @Test
    public void testScale2d1() throws Exception {

    }

    @Test
    public void testGetRandomInt() throws Exception {
        for(int i = 0; i < 30000; ++i) {
            int r = MyMathUtils.getRandomInt(1, 5);
//            log.v("getRandomInt(1, 5): %d", r);
            assertThat(r).isAtLeast(1);
            assertThat(r).isLessThan(5);
        }
        for(int i = 0; i < 30000; ++i) {
            int r = MyMathUtils.getRandomInt(-10000, 0);
//            log.v("getRandomInt(-10000, 0): %d", r);
            assertThat(r).isAtLeast(-10000);
            assertThat(r).isLessThan(0);
        }
    }

    @Test
    public void testGetRandomPoint() throws Exception {

    }

    @Test
    public void testScale0d1() throws Exception {

    }

    @Test
    public void testScale1d1() throws Exception {

    }

    @Test
    public void testScale2d2() throws Exception {

    }

    @Test
    public void testScale2d3() throws Exception {

    }

    @Test
    public void testGetRandomFloat() throws Exception {

    }

    @Test
    public void testGetRandomPointF() throws Exception {

    }

    @Test
    public void testGetRandomColor() throws Exception {

    }
}

File: myutils/src/test/java/pl/mareklangiewicz/myutils/MyMathUtilsTest.kt
package pl.mareklangiewicz.myutils

import org.junit.Test
import pl.mareklangiewicz.upue.Assert
import pl.mareklangiewicz.upue.IsAtLeast
import pl.mareklangiewicz.upue.IsLessThan
import pl.mareklangiewicz.upue.That

/**
 * Created by Marek Langiewicz on 04.11.15.
 */
class MyMathUtilsTest {

    @Test fun testGetRandomInt() {
        for (i in 0..29999) {
            val r = RANDOM.nextInt(1, 5)
            Assert That r IsAtLeast 1
            Assert That r IsLessThan 5
        }
        for (i in 0..29999) {
            val r = RANDOM.nextInt(-10000, 0)
            Assert That r IsAtLeast -10000
            Assert That r IsLessThan 0
        }
    }

    @Test fun testRandomAsInts() {
        val ints = RANDOM.asInts(1, 5)
        for (i in 0..29999) {
            val r = ints(Unit)
            Assert That r IsAtLeast 1
            Assert That r IsLessThan 5
        }
    }

    @Test fun testRandomAsFloats() {
        val floats = RANDOM.asFloats(1.1f, 5.5f)
        for (i in 0..29999) {
            val r = floats(Unit)
            Assert That r IsAtLeast 1.1f
            Assert That r IsLessThan 5.5f
        }
    }

    @Test fun testRandomAsDoubles() {
        val floats = RANDOM.asDoubles(1.1, 5.5)
        for (i in 0..29999) {
            val r = floats(Unit)
            Assert That r IsAtLeast 1.1
            Assert That r IsLessThan 5.5
        }
    }

    // can not test random points and colors here - it is android stuff...
}